### PR TITLE
CIV-4233 Yours Details Sole Trader

### PR DESCRIPTION
**Pull request by hmcts-version1-domini**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4233

### Change description ###
develop the 'Enter your details - Sole Trader' screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/urls.ts [hunk 1]
```diff
@@ -148,5 +148,7 @@ export const DQ_DEFENDANT_CAN_STILL_EXAMINE_URL = `${DIRECTIONS_QUESTIONNAIRE_UR
 export const DQ_SENT_EXPERT_REPORTS_URL = `${DIRECTIONS_QUESTIONNAIRE_URL}/sent-expert-reports`;
 export const DQ_OTHER_WITNESSES_AVAILABILITY_DATES_FOR_HEARING_URL = `${DIRECTIONS_QUESTIONNAIRE_URL}/availability-dates`;
 export const DQ_EXPERT_GUIDANCE_URL = `${DIRECTIONS_QUESTIONNAIRE_URL}/expert-guidance`;
+export const CLAIM_CLAIMANT_SOLE_TRADER_DETAILS_URL = `${BASE_CLAIM_URL}/claimant-sole-trader-details`;
```
Comment (laarmada) [inline src/main/routes/urls.ts:151]: why you add CLAIM_ ??? we already export const CLAIMANT_SOLE_TRADER_DETAILS_URL = `${BASE_CLAIM_URL}/claimant-sole-trader-details`;

Comment (laarmada) [reply src/main/routes/urls.ts:151]: why we need two?

File: src/main/routes/urls.ts [hunk 2]
```diff
@@ -148,5 +148,7 @@ export const DQ_DEFENDANT_CAN_STILL_EXAMINE_URL = `${DIRECTIONS_QUESTIONNAIRE_UR
...
+export const CLAIMANT_PHONE_URL = `${BASE_CLAIM_URL}/claimant-phone`;
```
Comment (laarmada) [inline src/main/routes/urls.ts:152]: not is the same???? export const CLAIMANT_PHONE_NUMBER_URL = `${BASE_CLAIM_URL}/claimant-phone-number`;

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/test/unit/routes/features/claim/claimantSoleTraderDetailsController.test.ts [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+jest.mock('../../../../../main/modules/draft-store/draftStoreService');
+jest.mock('../../../../../main/services/features/response/citizenDetails/citizenDetailsService');
+
+const mockGetCaseData = draftStoreService.getCaseDataFromStore as jest.Mock;
+const mockSaveDraftClaim = draftStoreService.saveDraftClaim as jest.Mock;
+
+const claim = new Claim();
+
+const buildClaimOfApplicant = (): Party => {
+  claim.applicant1 = new Respondent();
```
Comment (TatsyMaslova) [inline src/test/unit/routes/features/claim/claimantSoleTraderDetailsController.test.ts:32]: perhaps we should not create a new respondent there as claim.applicant is type of party

Comment (TatsyMaslova) [reply src/test/unit/routes/features/claim/claimantSoleTraderDetailsController.test.ts:32]: instead we can create populated party interface and assign it to claim.applicant1

File: src/test/unit/routes/features/claim/claimantSoleTraderDetailsController.test.ts [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+      .get(CLAIM_CLAIMANT_SOLE_TRADER_DETAILS_URL)
+      .expect((res) => {
+        expect(res.status).toBe(200);
+        expect(res.text).toContain('Enter your details');
+      });
+  });
+
+  it('should return your details page with information without correspondent address', async () => {
+    const buildClaimOfApplicantWithoutCorrespondent = (): Party => {
+      claim.applicant1 = new Respondent();
```
Comment (TatsyMaslova) [inline src/test/unit/routes/features/claim/claimantSoleTraderDetailsController.test.ts:129]: same comment as above

**Pull request closed without merging**